tests/integration-tests/conftest.py

```diff
@@ -312,6 +313,60 @@ def _cluster_factory(cluster_config, extra_args=None, raise_on_error=True):
         factory.destroy_all_clusters(test_passed=request.node.rep_call.passed)
 
 
+@pytest.fixture(scope='class')
+def api_server_factory(cfn_stacks_factory, request, public_ecr_image_uri, api_definition_s3_uri):
+    """Creates a factory for deploying API servers on-demand to each region."""
+    api_servers = {}
+
+    def _api_server_factory(server_region):
+        api_stack_name = generate_stack_name("integ-tests-api", request.config.getoption("stackname_suffix"))
+        stack_template_path = os.path.join("..", "..", "api", "infrastructure", "parallelcluster-api.yaml")
+        with open(stack_template_path) as stack_template_file:
+            stack_template_data = stack_template_file.read()
+
+        params = [{"ParameterKey": "PublicEcrImageUri", "ParameterValue": public_ecr_image_uri},
+                  {"ParameterKey": "ApiDefinitionS3Uri", "ParameterValue": api_definition_s3_uri}]
+
+        if server_region not in api_servers:
+            logging.info(f"Creating API Server stack: {api_stack_name} in {server_region}")
+            try:
+                set_credentials(server_region, request.config.getoption("credential"))
+                stack = CfnStack(name=api_stack_name,
+                                 region=server_region,
+                                 parameters=params,
+                                 capabilities=['CAPABILITY_IAM', 'CAPABILITY_AUTO_EXPAND'],
+                                 template=stack_template_data)
+                cfn_stacks_factory.create_stack(stack)
+            finally:
+                unset_credentials()
+            api_servers[server_region] = stack
+        else:
+            logging.info(f"Found cached API Server stack: {api_stack_name} in {server_region}")
+
+        return api_servers[server_region]
+
+    yield _api_server_factory
+
+
+@pytest.fixture(scope="class")
+def api_client(region, api_server_factory, api_uri):
+    """Define a fixture for an API client that interacts with the
+    pcluster api."""
+    from pcluster_client.api import cluster_operations_api
+    from pcluster_client import Configuration, ApiClient
+
+    if api_uri:
+        host = api_uri
+    else:
+        stack = api_server_factory(region)
+        host = stack.cfn_outputs["ParallelClusterApiInvokeUrl"]
+
+    api_configuration = Configuration(host=host)
+
+    with ApiClient(api_configuration) as api_client_instance:
+        yield cluster_operations_api.ClusterOperationsApi(api_client_instance)
```

Comment:
do you plan to return api_client_instance at some point to test endpoints that are not part of the ClusterOperationsApi?

Reply:
Yes --  I sent DM separately, but figured this could be an exercise of getting things end-to-end and expanded to provide the various clients (especially as the first round is iterated on). It's possible to:
- add the other clients here as well
- just yield the `api_client_instance` and then create the `cluster_operations_api` from within the test.

Reply:
Upon further thought, the second option feels better, at least for now so I pushed that change.